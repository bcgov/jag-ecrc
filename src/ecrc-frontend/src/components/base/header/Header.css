.container-fluid {
  width: 90%;
}
