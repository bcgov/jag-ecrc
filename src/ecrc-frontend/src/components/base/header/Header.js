import React from "react";
import PropTypes from "prop-types";

export default function Header({ header: { name } }) {
  return (
    <header>
      <nav className="navbar navbar-expand-lg navbar-dark">
        <div className="container">
          <a className="navbar-brand" href="/">
            <img
              className="img-fluid d-none d-md-block"
              src="/images/bcid-logo-rev-en.svg"
              width="181"
              height="44"
              alt="B.C. Government Logo"
            />
            <img
              className="img-fluid d-md-none"
              src="/images/bcid-symbol-rev.svg"
              width="64"
              height="44"
              alt="B.C. Government Logo"
            />
          </a>
          <div>
            <a className="navbar-brand nav-item nav-link" href="/">
              {name}
            </a>
          </div>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarNavAltMarkup"
            aria-controls="navbarNavAltMarkup"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon" />
          </button>
          <div className="collapse navbar-collapse" id="navbarNavAltMarkup" />
        </div>
      </nav>
    </header>
  );
}

Header.propTypes = {
  header: PropTypes.shape({
    name: PropTypes.string.isRequired
  }).isRequired
};
