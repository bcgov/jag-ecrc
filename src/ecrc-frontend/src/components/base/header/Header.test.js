import React from "react";
import { create } from "react-test-renderer";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { render, fireEvent, getAllByRole, wait } from "@testing-library/react";

import Header, { goHome } from "./Header";

describe("Header Component", () => {
  const header = {
    name: "eCRC"
  };

  window.confirm = jest.fn();

  test("Matches the snapshot", () => {
    const history = createMemoryHistory();

    const headerComponent = create(
      <Router history={history}>
        <Header header={header} />
      </Router>
    );
    expect(headerComponent.toJSON()).toMatchSnapshot();
  });

  test("Clicking header icon from a screen such as consent opens a confirmation popup", async () => {
    const history = createMemoryHistory();

    history.location.pathname = "/criminalrecordcheck/consent";

    const { container } = render(
      <Router history={history}>
        <Header header={header} />
      </Router>
    );

    fireEvent.click(getAllByRole(container, "button")[0]);

    await wait(() => {
      expect(window.confirm).toHaveBeenCalledWith(
        "You are in the middle of completing your eCRC. If you leave, your changes will be lost. Are you sure you would like to leave?"
      );
    });
  });

  test("Clicking header text from a screen such as success opens a confirmation popup", async () => {
    const history = createMemoryHistory();

    history.location.pathname = "/criminalrecordcheck/success";

    const { container } = render(
      <Router history={history}>
        <Header header={header} />
      </Router>
    );

    fireEvent.click(getAllByRole(container, "button")[1]);

    await wait(() => {
      expect(window.confirm).toHaveBeenCalledWith(
        "Are you sure you would like to leave this page?"
      );
    });
  });

  test("Key down on header icon from a screen such as consent opens a confirmation popup", async () => {
    const history = createMemoryHistory();

    history.location.pathname = "/criminalrecordcheck/consent";

    const { container } = render(
      <Router history={history}>
        <Header header={header} />
      </Router>
    );

    fireEvent.keyDown(getAllByRole(container, "button")[0]);

    await wait(() => {
      expect(window.confirm).toHaveBeenCalledWith(
        "You are in the middle of completing your eCRC. If you leave, your changes will be lost. Are you sure you would like to leave?"
      );
    });
  });

  test("Key down on header text from a screen such as consent opens a confirmation popup", async () => {
    const history = createMemoryHistory();

    history.location.pathname = "/criminalrecordcheck/consent";

    const { container } = render(
      <Router history={history}>
        <Header header={header} />
      </Router>
    );

    fireEvent.keyDown(getAllByRole(container, "button")[1]);

    await wait(() => {
      expect(window.confirm).toHaveBeenCalledWith(
        "You are in the middle of completing your eCRC. If you leave, your changes will be lost. Are you sure you would like to leave?"
      );
    });
  });

  test("Go Home function works as expected", () => {
    const event = {
      preventDefault: jest.fn(),
      stopPropagation: jest.fn()
    };
    const history = createMemoryHistory();

    history.location.pathname = "/criminalrecordcheck";

    jest.spyOn(event, "preventDefault");
    jest.spyOn(event, "stopPropagation");

    let result = goHome(event, history);

    expect(event.preventDefault).toBeCalled();
    expect(event.stopPropagation).toBeCalled();
    expect(result).toEqual(false);

    history.location.pathname = "/criminalrecordcheck/error";

    result = goHome(event, history);

    expect(result).toEqual(true);
    expect(history.location.pathname).toEqual("/");
    expect(sessionStorage.length).toEqual(0);

    history.location.pathname = "/criminalrecordcheck/consent";
    window.confirm = () => {
      return true;
    };

    result = goHome(event, history);

    expect(result).toEqual(true);
    expect(history.location.pathname).toEqual("/");
    expect(sessionStorage.length).toEqual(0);
  });
});
