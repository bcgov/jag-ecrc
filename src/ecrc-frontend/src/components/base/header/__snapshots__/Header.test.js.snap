// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Matches the snapshot 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-dark"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <img
          alt="B.C. Government Logo"
          className="img-fluid d-none d-md-block"
          height="44"
          src="/images/bcid-logo-rev-en.svg"
          width="181"
        />
        <img
          alt="B.C. Government Logo"
          className="img-fluid d-md-none"
          height="44"
          src="/images/bcid-symbol-rev.svg"
          width="64"
        />
      </a>
      <div>
        <a
          className="navbar-brand nav-item nav-link"
          href="/"
        >
          eCRC
        </a>
      </div>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarNavAltMarkup"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      />
    </div>
  </nav>
</header>
`;
